src/managed_data/_api.py

```diff
@@ -77,7 +77,7 @@ def list_files(
     assigned_row_ids: Optional[list[str]] = None,
     is_unassigned: Optional[bool] = None,
     client: Optional[Client] = None,
-):
+) -> list[dict]:
```

Comment:
Possibly in a future issue/PR, can we use the `schema` module to type this? This will make it easier to understand the structure of the output.

Reply:
let's talk about this soon -- i'm not happy with the way things are.

Reply:
Perhaps I'm missing something. We should be able to use Pydantic, dataclasses, or custom classes to model the output of each function. This might require more classes.

In any case, its not critical. We can make stricter typing a goal for a second enhanced and more performant version.